apps/cowswap-frontend/src/modules/twap/hooks/useEmulatedTwapOrders.ts

```diff
@@ -32,10 +40,12 @@ export function useEmulatedTwapOrders(tokensByAddress: TokensByAddress | undefin
         return acc
       }
 
-      const orderDetails = mapTwapOrderToStoreOrder(order, tokensByAddress)
-      if (orderDetails) {
-        acc.push(orderDetails)
+      try {
+        acc.push(mapTwapOrderToStoreOrder(order, tokensByAddress))
+      } catch (e) {
+        console.error(e, order)
```

Comment:
im not sure this is the way.

My concern is. This race condition is normal. For a normal case, we are creating a lot of noise, and we might be covering up some other errors. what if `map` has any other error we need to address? 


Imo, the issue is how we model the token information. We use:

```ts
export interface TokensByAddress {
  [address: string]: TokenWithLogo
}
```

This has this effect:
<img width="898" alt="image" src="https://github.com/cowprotocol/cowswap/assets/2352112/bd01a0f0-ca0b-4737-b064-3f4122f3a741">

If we use a `Map` for example, the `get` will force you to handle the undefined case. 

Alternativelly, we need we could just add:
```ts
export interface TokensByAddress {
  [address: string]: TokenWithLogo | undefined
}
```

Reply:
Thanks!
Refactored: https://github.com/cowprotocol/cowswap/pull/3470/commits/c9235d21a20ba0c0181c0f0120ea3dd6f5f84014

Reply:
Thanks!!